doc/syslog/syslog-rate-limit-design.md

```diff
@@ -0,0 +1,383 @@
+{% if 'rate_limit_interval' in SYSLOG_CONFIG_FEATURE[container_name]%}
+{% set rate_limit_interval = SYSLOG_CONFIG_FEATURE[container_name]['rate_limit_interval'] %}
+{% endif %}
+{% if 'rate_limit_burst' in SYSLOG_CONFIG_FEATURE[container_name]%}
+{% set rate_limit_burst = SYSLOG_CONFIG_FEATURE[container_name]['rate_limit_burst'] %}
+{% endif %}
+{% endif %}
+{% endif %}
+
+{% if rate_limit_interval is defined %}
+$SystemLogRateLimitInterval {{ rate_limit_interval }}
+{% endif %}
+{% if rate_limit_burst is defined %}
+$SystemLogRateLimitBurst {{ rate_limit_burst }}
+{% endif %}
+```
+
+#### docker_image_ctl.j2
+
+Function `updateSyslogConf` in the template currently only works for multi ASIC platform, it shall be extended to work for both multi ASIC and single ASIC platform. The only difference between multi ASIC and single ASIC platform is the target IP of rsyslog.
+
+- Mulit ASIC: target IP of rsyslog is get from docker0 IP which is already implemented in docker_image_ctl.j2
+- Single ASIC: target IP of rsyslog is always 127.0.0.1.
+
+Meanwhile, the file https://github.com/Azure/sonic-buildimage/blob/master/dockers/docker-base/etc/rsyslog.conf as well as any code processing it shall be removed.
+
+#### APP extension
+
+APP extension shall be able to expose its constant syslog capability by adding new fields to manifest:
+
+```
+root@sonic:/home/admin# spm show package manifest what-just-happened
+{
+    ...
+    "service": {
+        ...
+        "syslog": {
+            "support-rate-limit": "true"
+        }
+        ...
+    }
+    ...
+}
+```
+
+- support-rate-limit: indicates if this APP extension supports configuring rate limit. This field affected CLI behavior: if true, CLI shall put rate limit configuration to CONFIG DB; otherwise, CLI shall reject the configuration.
+
+The capability shall also be saved to CONFIG DB to allow easy access by management plane:
+
+```
+root@sonic:/home/admin# cat /etc/sonic/config_db.json
+{
+    "FEATURE": {
+        ...
+        "what-just-happened": {
+            ...
+            "support_syslog_rate_limit": "True",
+            ...
+        }
+        ...
+}
+```
+
+Also, default value shall be provided to https://github.com/Azure/sonic-utilities/blob/master/sonic_package_manager/service_creator/feature.py:
+
+```
+DEFAULT_SYSLOG_FEATURE_CONFIG = {
+    'rate_limit_interval': '300',
+    'rate_limit_burst': '20000'
+}
+```
+
+### SAI API
+
+N/A
+
+### Configuration and management
+
+#### CLI change
+
+Config rate limit:
+
+```
+config syslog rate-limit-host --interval <interval> --burst <burst>
+config syslog rate-limit-container <service_name> --interval <interval> --burst <burst>
+
+Example:
+config syslog rate-limit-host --interval 300 --burst 20000
+config syslog rate-limit-host --interval 300
+config syslog rate-limit-host --burst 20000
+config syslog rate-limit-container bgp --interval 300 --burst 20000
+```
+
+> Note: set interval or burst to 0 will disable rate limit.
+
+Show rate limit:
```

Comment:
@Junchao-Mellanox why not to have like this:
```
show
|--- syslog
     |--- rate-limit
          |--- host
          |--- container <service_name>
```

Reply:
Good suggestion. But I don't see an obviously benefit comparing to the current design. As the code has been implemented for a long time, I would like to keep it as `rate-limit-host` and `rate-limit-container`

Reply:
@Junchao-Mellanox ok. Got it.